code/datums/components/_component.dm

```diff
@@ -19,6 +19,8 @@
 
 /datum/component/proc/_JoinParent()
 	var/datum/P = parent
+	if(!P)
```

Comment:
if some component's being assigned to a null parent, investigate that. don't just do this, i'm sure alot of things in the component uses the parent var

Reply:
Right, I should look into this deeper.